### PR TITLE
Ruby kafka gem instrumentation

### PR DESCRIPTION
**Pull request by tannalynn**
Adds instrumentation for the ruby-kafka gem.

full ci run: https://github.com/newrelic/newrelic-ruby-agent/actions/runs/10795710672

### CODE REVIEW

File: lib/new_relic/agent/instrumentation/ruby_kafka/instrumentation.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    MESSAGING_LIBRARY = 'Kafka'
+    PRODUCE = 'Produce'
+    CONSUME = 'Consume'
+
+    INSTRUMENTATION_NAME = 'ruby-kafka'
+
+    def produce_with_new_relic(value, **kwargs)
+      NewRelic::Agent.record_instrumentation_invocation(INSTRUMENTATION_NAME)
+
+      topic_name = kwargs[:topic]
```
[the remark this answers is not in the document]
Comment (tannalynn) [reply lib/new_relic/agent/instrumentation/ruby_kafka/instrumentation.rb:18]: Fortunately, topic is actually a required keyword, so it's guaranteed to always be there
https://github.com/zendesk/ruby-kafka/blob/master/lib/kafka/producer.rb#L195

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,11 +2,13 @@
 
 ## dev
 
-Version <dev> adds instrumentation for the rdkafka gem.
+Version <dev> adds kafka instrumentation for the rdkafka and ruby-kafka gems.
```
Comment (kaylareopelle) [inline CHANGELOG.md:5]:
```suggestion
Version <dev> adds Apache Kafka instrumentation for the rdkafka and ruby-kafka gems.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,11 +2,13 @@
...
+
+- **Feature: Add kafka instrumentation for the rdkafka and ruby-kafka gems**
```
Comment (kaylareopelle) [inline CHANGELOG.md:7]:
```suggestion
- **Feature: Add Apache Kafka instrumentation for the rdkafka and ruby-kafka gems**
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,11 +2,13 @@
...
+
+  The agent now has instrumentation for both the rdkafka and ruby-kafka gems. The agent will now record transactions and message broker segments for produce and consume calls made using these gems. [PR#2824](https://github.com/newrelic/newrelic-ruby-agent/pull/2824) [PR#2842](https://github.com/newrelic/newrelic-ruby-agent/pull/2842)
```
Comment (kaylareopelle) [inline CHANGELOG.md:9]:
```suggestion
  The agent now has instrumentation for both the rdkafka and ruby-kafka gems. The agent will record transactions and message broker segments for produce and consume calls made using these gems. [PR#2824](https://github.com/newrelic/newrelic-ruby-agent/pull/2824) [PR#2842](https://github.com/newrelic/newrelic-ruby-agent/pull/2842)
```

**Review by kaylareopelle — COMMENTED** [review]
Few small docs things. I agree with @fallwith's suggestion to lock down the versions tested in the CI.

Looks good otherwise! Thank you!

File: test/multiverse/suites/ruby_kafka/Envfile [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+instrumentation_methods :chain, :prepend
+
+gemfile <<~RB
```
[the remark this answers is not in the document]
Comment (tannalynn) [reply test/multiverse/suites/ruby_kafka/Envfile:12]: ok added a suite condition here https://github.com/newrelic/newrelic-ruby-agent/pull/2842/commits/14b23d272a5d57dc2971698a4a3e144324e5dd6b
so now it'll run locally on any version, but CI will only run it on < 3.3

File: test/multiverse/suites/ruby_kafka/Envfile [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+suite_condition('Skip in CI on newer ruby versions') do
+  # will run locally OR on CI for ruby < 3.3.0
+  !ENV['CI'] || RUBY_VERSION < '3.3.0'
```
Comment (fallwith) [inline test/multiverse/suites/ruby_kafka/Envfile:7]:
```suggestion
  # will run locally OR on CI for ruby < 3.4.0
  !ENV['CI'] || RUBY_VERSION < '3.4.0'
```

I love the `suite_condition` idea. I think the range should include Ruby v3.3.0, as it exists at the time the instrumentation is being added.

**Review by fallwith — APPROVED** [review]

**Review by kaylareopelle — APPROVED** [review]
🦄

**Pull request merged**